File: Reputaly.Api/Domain/Review.cs
namespace Reputaly.API.Domain;

public class Review
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid TenantId { get; set; }
    public Tenant Tenant { get; set; } = null!;

    public Guid LocationId { get; set; }
    public TenantLocation Location { get; set; } = null!;

    // ID único de la reseña en Google — evita duplicados en el polling
    public string GoogleReviewId { get; set; } = string.Empty;

    public string AuthorName { get; set; } = string.Empty;

    // 1 a 5 estrellas
    public int Rating { get; set; }

    // Nullable: hay reseñas de Google que solo tienen estrellas, sin texto
    public string? Content { get; set; }

    public DateTime PublishedAt { get; set; }

    public string? DetectedLanguage {  get; set; }

    // Analisis de IA
    public decimal? SentimentScore {  get; set; }  // -1.0 a 1.0
                                                   
    public string[]? DetectedTopics {  get; set; }  // ["precio","trato","limpieza"]
    public DateTime? AiAnalyzedAt {  get; set; }



    // pending | auto_replied | escalated | replied
    public string Status { get; set; } = "pending";

    // Lo que la IA sugiere responder
    public string? AiSuggestedReply { get; set; }

    // auto_reply | escalate
    public string? AiDecision { get; set; }

    // Por qué la IA tomó esa decisión (útil para debugging y transparencia)
    public string? AiDecisionReason { get; set; }

    // La respuesta final publicada en Google (puede ser la de la IA o una manual)
    public string? FinalReply { get; set; }

    public DateTime? RepliedAt { get; set; }
    public DateTime? EscalatedAt { get; set; }

    // True si la respuesta se publicó automáticamente sin intervención humana
    public bool AutoReplied { get; set; } = false;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: reputaly-web/src/types/review.ts
export type ReviewStatus =
| 'pending'
| 'auto_replied'
| 'escalated'
| 'replied'
| 'ignored';

// Espejo del ReviewDTo del backend.
// Los campos son | null corresponden a os nullable de c#
// las fechs llegan como string en formato ISO
export interface Review {
    id: string;
    authorName: string;
    rating: number;
    content: string;
    publishedAt: string;
    status: ReviewStatus;
    aiSuggestedReply: string | null;
    aiDecision : string | null;
    aiDecisionReason: string | null;
    finalReply: string | null;
    repliedAt: string | null;
    escalatedAt: string | null;
    createdAt: string;
    locationId: string;
    detectedLanguage: string | null;
    sentimentScore: number | null;
    detectedTopics: string[] | null;
    autoReplied:boolean;
}

export interface ReviewsPage{
    total: number;
    page: number;
    pageSize: number;
    items: Review[];
}